### DISCUSSION TITLE
Action labels to support multiple environments

### DISCUSSION THREAD

**Discussion by jpwilliams** [interval/interval-node#4] [Ideas]
Hi folks,

We have a few different environments utilised by other teams and were looking at elegant ways of providing a set of actions for each of those environments.

We've just settled on naming, i.e. `Service A: Do thing` or `Service A (staging): Do thing`. It works, and filtering the actions by `staging` gets you a list of all actions in that environment, but the small addition of labels here feels like it'd be great for discovery.

For example, in my actions above, I might add the label "Service A" and "Staging". Having these labels then appear in the UI and allowing filtering based on them feels like it would provide an excellent experience for new team members discovering and using actions.

As before - ace tool! Hope I can provide some helpful insight into usage.

**Comment by alexarena** [comment]
Hey Jack, love this idea! For the sake of simplicity, we cut environments from our initial release, but this is something we'd like to add soon. Happy you found a workaround now + we'll let you know as soon as we have a more concrete proposal on how we plan to accomplish this. Would love to get your feedback when that happens.

**Reply by ChrisTalman** [reply]
This would be super useful.

At a certain threshold, the number of actions displayed on the dashboard can become a bit overwhelming to look through.

Having some way to breakdown actions into categories, like a label and filter system, would be very helpful.

**Reply by alexarena** [reply]
Great timing for this comment. We're still working on formal "environments" support but we've actually just shipped our beta for nested actions which lets you define (in code) a hierarchy for your actions. We can get you into this beta if you like!

More info here: https://interval.com/blog/weekly-update-003#-nested-actions-enters-beta

**Reply by ChrisTalman** [reply]
Sure, that'd be cool. 🙂
